ethereum/ts-scripts/relayer/deliveryProvider/upgradeDeliveryProvider.ts

```diff
@@ -0,0 +1,40 @@
+import {
+  init,
+  writeOutputFiles,
+  ChainInfo,
+  Deployment,
+  getDeliveryProvider,
+  getOperatingChains,
+} from "../helpers/env";
+import { deployDeliveryProviderImplementation } from "../helpers/deployments";
+
+const processName = "upgradeDeliveryProvider";
+init();
+const chains = getOperatingChains();
+
+async function run() {
+  console.log("Start!");
+  const output: any = {
+    deliveryProviderImplementations: [],
+  };
+
+  for (let i = 0; i < chains.length; i++) {
+    const deliveryProviderImplementation = await deployDeliveryProviderImplementation(
+      chains[i]
+    );
+    await upgradeDeliveryProvider(chains[i], deliveryProviderImplementation);
+
+    output.deliveryProviderImplementations.push(deliveryProviderImplementation);
+  }
+
+  writeOutputFiles(output, processName);
+}
+
+async function upgradeDeliveryProvider(chain: ChainInfo, newImpl: Deployment) {
+  console.log("About to upgrade relay provider for chain " + chain.chainId);
+  const provider = getDeliveryProvider(chain);
+  await provider.upgrade(chain.chainId, newImpl.address);
```

Comment:
Should we fetch the receipt from the transaction with `tx.wait()` to wait for a confirmation?